src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1547,5 +1547,24 @@ internal bool Equals(TypeSymbolWithAnnotations other)
         {
             throw ExceptionUtilities.Unreachable;
         }
+
+        public static bool Equals(TypeSymbol left, TypeSymbol right, TypeCompareKind comparison)
+        {
+            if (comparison == TypeCompareKind.ConsiderEverything && ((object)right == null))
+            {
+                return (object)left == (object)null;
+            }
+
+            return (comparison == TypeCompareKind.ConsiderEverything && (object)left == (object)right)
+                || right.Equals(left, comparison);
+        }
+
+        [Obsolete("Use TypeSymbol.Equals method.", true)]
```

Comment:
>TypeSymbol.Equals [](start = 23, length = 17)

It might be worth spelling out the signature to make it easier to locate the API